elyra/pipeline/registry.py

```diff
@@ -0,0 +1,483 @@
+#
+# Copyright 2018-2020 IBM Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import os
+import io
+import json
+import yaml
+import urllib
+
+from traitlets.config import SingletonConfigurable
+
+
+cardinality = {
+    'min': 0,
+    'max': -1
+}
+
+inputs = {
+    "id": "inPort",
+    "app_data": {
+        "ui_data": {
+            "cardinality": cardinality,
+            "label": "Input Port"
+        }
+    }
+}
+
+outputs = {
+    "id": "outPort",
+    "app_data": {
+        "ui_data": {
+            "cardinality": cardinality,
+            "label": "Output Port"
+        }
+    }
+}
+
+default_current_parameters = {
+    "filename": "",
+    "runtime_image": "",
+    "outputs": [],
+    "env_vars": [],
+    "dependencies": [],
+    "include_subdirectories": False
+}
+
+
+def get_id_from_name(name):
+    return ' '.join(name.lower().replace('-', '').split()).replace(' ', '-').replace('_', '-')
```

Comment:
Please add a docstring indicating what this is doing and why it's necessary.  The removal of `-` in the original, then its re-introduction for space and underscores is a little funky.